tfhe/src/shortint/engine/server_side/mod.rs

```diff
@@ -681,11 +681,15 @@ impl ShortintEngine {
         server_key: &ServerKey,
         ct: &mut Ciphertext,
     ) -> EngineResult<()> {
-        let modulus = ct.message_modulus.0 as u64;
+        if ct.carry_is_empty() {
+            self.create_trivial_assign(server_key, ct, 0)?;
```

Comment:
This change is dangerous as it currently is also used for noise management, with a degree/noise rework yes, but I would not make that change/check currently

Reply:
I agree
I'll keep this commit in another branch for later

Reply:
Done